src/Network/Mattermost/Types/Internal.hs

```diff
@@ -83,6 +83,7 @@ data ConnectionData
   = ConnectionData
   { cdHostname       :: Hostname
   , cdPort           :: Port
+  , cdDirectory      :: String
```

Comment:
Would you be willing to rename this (and associated variable bindings accordingly) to `cdUrlPath`? Although a directory might be involved on the server side, from the API library's perspective the consequence is that the connection needs to be made with respect to a specific URL path rather than a directory.